pallets/mint_currency/src/lib.rs

```diff
@@ -0,0 +1,115 @@
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+pub use pallet::*;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use frame_support::{
+        dispatch::DispatchResult, pallet_prelude::*,
+        traits::Currency,
+        sp_runtime::traits::CheckedConversion
+    };
+	use frame_system::{
+        pallet_prelude::*,
+        ensure_root,
+    };
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// Because this pallet emits events, it depends on the runtime's definition of an event.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+        type Currency: Currency<Self::AccountId>;
+	}
+
+    // type BalanceOf<T> = <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+    // type NegativeImbalanceOf<T> = <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::NegativeImbalance;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	// #[pallet::storage]
+	// #[pallet::getter(fn something)]
+	// pub type Something<T> = StorageValue<_, u32>;
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// the proposal value have been minted on the target account
+		/// [proposal_id, target_account, value]
+		ProposalValueMinted(u32, T::AccountId, u128),
+		/// the proposal value have been added on the target account from the treasury
+		/// [proposal_id, target_account, value]
+		ProposalValueAddedFromTreasury(u32, T::AccountId, u128),
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The proposal have not been yet accepted
+		ProposalNotAccepted,
+        /// There is not enough funds in the treasury to pay the proposal
+        NotEnoughFunds,
+        /// Overflow when adding the value to the account
+		Overflow,
+        /// TO DELETE ON PROD
+        NotImplementedYet,
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::weight(10_000)]
+		pub fn proposal_value_mint(
+            origin: OriginFor<T>,
+			proposal_hash: u32,
+			target_account: T::AccountId,
+			network_service_provider: T::AccountId,
+            amount: u128,
+		) -> DispatchResult {
+            // should the transaction be signed ?
+            // if yes, by root ?
+            ensure_root(origin)?;
+
+            // ensure the proposal_hash have been accepted
+            // else, return Error::ProposalNotAccepted
+
+            let fee_amount: u128 = amount / 10;
+
+            let neg_imbalance = T::Currency::issue(amount.checked_into().ok_or(Error::<T>::Overflow)?);
+            let neg_fee_imbalance = T::Currency::issue(fee_amount.checked_into().ok_or(Error::<T>::Overflow)?);
+
+            if let Err(_) = T::Currency::resolve_into_existing(&target_account, neg_imbalance) {
```

Comment:
I think we should use `resolve_creating` instead of `resolve_into_existing`.
The beneficiary will often be a multisign account, created for the only purpose of receiving this money. It won't have any previous funds and must therefore be brought to existence during the transaction.

Reply:
Also it doesn't return any error.
Btw it does not fait in case of overflow. It fail if the account doesn't already exist. And it return the same imbalance as the error, still to be managed by the code logic

Reply:
Good point for the function.
For the ``deposit_into_existing`` i was using, in the source code of Balances, there was this line:
``account.free = account.free.checked_add(&value).ok_or(ArithmeticError::Overflow)?;``

that's why i made an overflow error, but ``resolve_creating()`` solve the problem

Reply:
The source code is not in Balance, but in currency trait: https://docs.substrate.io/rustdocs/latest/frame_support/traits/tokens/currency/trait.Currency.html
```
	/// Similar to deposit_creating, only accepts a `NegativeImbalance` and returns nothing on
	/// success.
	fn resolve_into_existing(
		who: &AccountId,
		value: Self::NegativeImbalance,
	) -> Result<(), Self::NegativeImbalance> {
		let v = value.peek();
		match Self::deposit_into_existing(who, v) {
			Ok(opposite) => Ok(drop(value.offset(opposite))),
			_ => Err(value),
		}
	}
```
What was the part you were reading ?

Reply:
`resolve_into_existing()` use `deposit_into_existing()`, that is implemented by Balance, and it's the only Err the implementation is returning, so the only case of failing for `resolve_into_existing` is an overflow, at least for Balance implementation, but i'm using a trait, not the implementation, i should overlook the error